### PR TITLE
BAU BAU removed unused safe id in domain object so only registeredBusinessPartnerId is used

### PR DESCRIPTION
**Pull request by paulblakehmrc**
Two safeIds are modelled in the Organisation domain object. Removed `safeNumber` at the top level and used the embedded registeredBusinessPartnerId for checks on whether a safeId is present or not

### CODE REVIEW

File: app/uk/gov/hmrc/plasticpackagingtax/registration/models/registration/OrganisationDetails.scala [hunk 1]
```diff
@@ -56,21 +55,23 @@ case class OrganisationDetails(
         details.registration.registrationStatus == "REGISTRATION_NOT_CALLED" && details.businessVerificationStatus == "FAIL"
     )
 
-  def businessPartnerIdPresent(): Boolean =
+  def businessPartnerId(): Option[String] =
     organisationType match {
       case Some(UK_COMPANY) =>
-        incorporationDetails.exists(_.registration.registeredBusinessPartnerId.isDefined)
+        incorporationDetails.fold[Option[String]](None)(_.registration.registeredBusinessPartnerId)
```
Comment (RyanKellyHMRC) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/models/registration/OrganisationDetails.scala:61]: I might be wrong but doesn't Option.map() do want you want here?

Comment (RyanKellyHMRC) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/models/registration/OrganisationDetails.scala:61]: Or maybe `flatMap()`?

Comment (SteveSugden) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/models/registration/OrganisationDetails.scala:61]: Agree with Ryan.   This should do the trick
```

  def businessPartnerId(): Option[String] =
    organisationType match {
      case Some(UK_COMPANY) =>
        incorporationDetails.flatMap(_.registration.registeredBusinessPartnerId)
      case Some(SOLE_TRADER) =>
        soleTraderDetails.flatMap(_.registration.registeredBusinessPartnerId)
      case GENERAL_PARTNERSHIP =>
        partnershipDetails.flatMap(
          _.generalPartnershipDetails.flatMap(_.registration.registeredBusinessPartnerId)
        )
      case SCOTTISH_PARTNERSHIP =>
        partnershipDetails.flatMap(
          _.scottishPartnershipDetails.flatMap(_.registration.registeredBusinessPartnerId)
        )
      case _ => None
    }
```

Comment (SteveSugden) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/models/registration/OrganisationDetails.scala:61]: Actually, that is not quite right.  Partnership needs splitting up as before

Comment (paulblakehmrc) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/models/registration/OrganisationDetails.scala]: Addressed with use of flatMap

**Review by RyanKellyHMRC — APPROVED** [review]

**Pull request merged**